sale_margin_delivered/models/sale_margin.py

```diff
@@ -31,6 +31,20 @@ class SaleOrderLine(models.Model):
         "Formula: Value Delivered / Quantity Delivered",
     )
 
+    def _get_delivered_margin_valuation_layers(self):
```

Comment:
@Shide we could opt for a more meaningful method name
```suggestion
    def _get_delivered_qty_and_value(self):
        """ Return the qty and the value of delivered products
        """
        ....
```

What do you think of also returning the qty delivered via this method. This way, we remove the use of valuation layers in the '_compute_margin_delivered' method and we make-it plug-gable...

Reply:
The quantity and value fields are always calculated the same based on the valuation_layers (sum).

Having the function that decides which Valuation Layers are going to be used, allows us to abstract from the rest of the flow and to be able to easily inherit in the future if we decide to add another use case (as I have done with Dropshipping).

If I added the Value and Quantity to the function, every time we inherited we would have to reselect all the Valuation Layers involved in calculation, and that would not be possible if there are several modules inheriting the same function.

TLDR; I have decided not to put it in the function because the calculation of the quantity and value is always the same and would not allow a simple inheritance.

I'm going to improve doc of the functions